src/ui/CommerceQuery/CommerceQuery.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { Assert } from '../../misc/Assert';
+import { QueryEvents, IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { AnalyticsEvents, IChangeAnalyticsCustomDataEventArgs } from '../../events/AnalyticsEvents';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface ICommerceQueryOptions {
+  listing?: string;
+}
+
+/**
+ * The CommerceQuery component enables to set properties for listing pages in commerce sites in a convenient way
+ */
+export class CommerceQuery extends Component {
+  static ID = 'CommerceQuery';
+
+  static doExport = () => {
+    exportGlobally({
+      CommerceQuery: CommerceQuery
+    });
+  };
+
+  /**
+   * The options for the CommerceQuery.
+   * @componentOptions
+   */
+  static options: ICommerceQueryOptions = {
+    /**
+     * Setting this property will update the tab value of a query and set the originLevel2 in the analytics event
+     *
+     * This is useful for listing pages in commerce sites to leverage the Coveo ML features.
+     */
+    listing: ComponentOptions.buildStringOption()
+  };
+
+  /**
+   * Creates a new CommerceQuery component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the CommerceQuery component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options?: ICommerceQueryOptions, public bindings?: IComponentBindings) {
+    super(element, CommerceQuery.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, CommerceQuery, options);
+    Assert.exists(element);
+    Assert.exists(this.options);
```

Comment:
I'm curious why are these assertions needed? I see that we assert the truthiness of the element in the `BaseComponent` base class. I also don't see other components asserting `this.options`? Could the problem be with `ComponentOptions.initComponentOptions` not always returning an object?

Reply:
I took `Triggers` component as a starting point for this component and it has those assertions. Also `DidYouMean` and `ResultList`. But good point, I don't think they're needed.